### DISCUSSION TITLE
hy.errors.HyWrapperError: Don't know how to wrap <class …>

### DISCUSSION THREAD

**Discussion by wrobell** [hylang/hy#2465] [Q&A] [answered]
Please consider the following script
```
(eval-when-compile
  (import dataclasses [dataclass])

  (defclass [dataclass] Column []
    #^str name
    #^str type)

  (defn f-column [name type] (Column name type)))

(defmacro m-create [name type] (f-column name type))
(print (m-create "help" "text"))
```
Running it I am getting the error
```
expanding macro m-create
  hy.errors.HyWrapperError: Don't know how to wrap <class 'm-test.Column'>: Column(name=hy.models.String('help'), type=hy.models.String('text'))
```

Any suggestions now to deal with this?

**Comment by Kodiologist** [comment]
As the error message implies, you've written a macro that returns a `Column` object. It's hard for me to see what Hy code a `Column` object should correspond to, so I'm not sure what you expected.

**Reply by wrobell** [reply]
I am writing something more complex, and I got such error. The example here is just to replicate the error I am getting.
I would expect the macro to create the object. Nothing more. What am I missing?

**Reply by Kodiologist** [reply]
The macro does create the object. At compile-time, when it's expanded. And then it tries to make source code (more specifically: a Hy model) out of that object, which it sounds like isn't what you want. Remember that fundamentally, a macro is code for generating code.

**Reply by wrobell** [reply]
Got it.. I hope ;)
How can I change function `f-column` to generate the model to create that object on execution?

**Reply by Kodiologist** [reply, marked as answer]
[Quasiquoting](https://docs.hylang.org/en/master/api.html#quasiquote) is usually the easiest way to do it.

**Reply by wrobell** [reply]
Thanks. For other lost souls

```
(import dataclasses [dataclass])
(defclass [dataclass] Column []
  #^str name
  #^str type)

(eval-when-compile
  (defn f-column [name type] `(Column ~name ~type)))

(defmacro m-create [name type] (f-column name type))
(print (m-create "help" "text"))
```
